### PR TITLE
Allow shared RandomState/Generator updates in JAX backend

### LINKED ISSUE #982: Update of random variables not working in JAX backend
```python
import aesara
import aesara.tensor as at
import numpy as np

rng = aesara.shared(np.random.default_rng())
x = at.random.normal(rng=rng)
f = aesara.function([], x, updates={rng: x.owner.outputs[0]}, mode="JAX")
assert f() != f()  # Fails
```

### PR DESCRIPTION
**Pull request by ricardoV94**
Closes #982

A more clean solution would be to replace the SharedRandomState/Generator variables by Shared JAX [PRNGKeys](https://jax.readthedocs.io/en/latest/_autosummary/jax.random.PRNGKey.html#jax.random.PRNGKey), but this works for now.

### CODE REVIEW

File: tests/link/jax/test_random.py [hunk 1]
```diff
@@ -79,8 +82,34 @@ def test_RandomStream():
...
     jax_res_1 = fn()
     jax_res_2 = fn()
 
-    assert np.array_equal(jax_res_1, jax_res_2)
+    assert not np.array_equal(jax_res_1, jax_res_2)
+
+
+@pytest.mark.parametrize("rng_ctor", (np.random.RandomState, np.random.default_rng))
+def test_random_updates(rng_ctor):
+    original_value = rng_ctor()
```
Comment (brandonwillard) [inline tests/link/jax/test_random.py:98]: Perhaps we should set a seed value here to guarantee the consistency of the `assert` that follows?

Comment (ricardoV94) [reply tests/link/jax/test_random.py:98]: The f() != f()? That should be incredibly rare to happen by chance.

Comment (brandonwillard) [reply tests/link/jax/test_random.py:98]: No chance is better than any in this scenario, especially given the effort required to avoid it entirely.

Comment (ricardoV94) [reply tests/link/jax/test_random.py]: Done

**Review by brandonwillard — APPROVED** [review]
